### ISSUE TITLE
Drivers issue on respeaker 4mic array

### ISSUE THREAD

**Issue by dave-parsons** [rhasspy/wyoming-satellite#116]
Using this HAT https://wiki.seeedstudio.com/ReSpeaker_4_Mic_Array_for_Raspberry_Pi/ 

Pi freezes and requires hard reboot when attempting record test per tutorial setup instructions. 

Resolved by reinstalling drivers per here after the "rebuild drivers" step in tutorial.  

sudo apt-get update
git clone https://github.com/HinTak/seeed-voicecard.git
cd seeed-voicecard
sudo ./install.sh
sudo reboot now

Successfully records and plays now.

**Comment by synesthesiam** [comment]
Do you get high CPU usage when recording with the 4mic HAT?

**Comment by dave-parsons** [comment]
No.   %CPU on Pi4 looks to stay in the 20-35% range.   I'm uncertain if it's actually using 4 mics or just 1, but it does work.

**Comment by nubicula** [comment]
Have either of you got this working with Bookworm? Whenever I try with Bookworm and perform the following command, it always end up with kernel panic.

arecord -D plughw:CARD=seeed4micvoicec,DEV=0 -r 16000 -c 1 -f S16_LE -t wav -d 5 test.wav

Unable to handle kernel paging request at virtual address .....
This is on a Raspberry Pi 4 Model B 1.1 with the 4-mic respeaker hat.

Only way to get this working so far was to use Bullseye. What are you both using, and potentially any tips for Bookworm?

**Comment by dave-parsons** [comment]
Sorry - no attempts or knowledge on bookworm.

**Comment by dave-parsons** [comment]
Tried the new installer on Pi3B+ with usb mic.  Worked flawlessly.   

I was so impressed I decided to install the installer on my working Pi4 with 4mic from above.   Appeared to be working fine but the hang reappeared.   Reinstalled drivers like above but didn't fix it.   I've been troubleshooting most of the night step by step through the manual setup.   I'm having sporadic success getting wake word, recording, and play back - but it's cranky and hangs.   LED's stopped working and haven't figured that out yet.   Thinking tomorrow start fresh with manual install again.

**Comment by dave-parsons** [comment]
Started with a clean image today.
Confirmed installer version of respeaker drivers didn't pickup devices.  
Had to install from here:  git clone https://github.com/HinTak/seeed-voicecard.git

Continued using installer to setup all other configurations.   All works except LEDs.  

Any advice on how to get 4mic LEDs working?

**Comment by dave-parsons** [comment]
> Have either of you got this working with Bookworm? Whenever I try with Bookworm and perform the following command, it always end up with kernel panic.
> 
> arecord -D plughw:CARD=seeed4micvoicec,DEV=0 -r 16000 -c 1 -f S16_LE -t wav -d 5 test.wav
> 
> Unable to handle kernel paging request at virtual address ..... This is on a Raspberry Pi 4 Model B 1.1 with the 4-mic respeaker hat.
> 
> Only way to get this working so far was to use Bullseye. What are you both using, and potentially any tips for Bookworm?

Now I see what you are saying.  I'm having the same issue.  It has to do with the driver.  I was able to get it to work with manual install - but no luck since trying the installer 6.1.   I'm still trying..

**Comment by llluis** [comment]
> Any advice on how to get 4mic LEDs working?

There's a pending PR with the code for it. 
Try it here: #126

**Comment by dave-parsons** [comment]
Spent most of the day trying to get 4mic HAT working on Pi4 with installer.   Just couldn't make it work.   

Reformatted and went back to manual install (using drivers install from [seek](https://github.com/HinTak/seeed-voicecard.git).   

All working - including LEDs on Bookworm 64bit.   Something's going on with installer that I don't understand, but my next try will be on a different SD card :)
